pkg/apis/ingress/v1alpha1/status.go

```diff
@@ -0,0 +1,190 @@
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	k8sutil "github.com/operator-framework/operator-sdk/pkg/util/k8sutil"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// setStatusCondition modifies the ClusterIngress object's status by setting the
+// specified condition.
+func (ci *ClusterIngress) setStatusCondition(condition *operatorapi.OperatorCondition) {
+	condition.LastTransitionTime = metav1.Now()
+
+	conditions := []operatorapi.OperatorCondition{}
+
+	found := false
+	for _, c := range ci.Status.Conditions {
+		if condition.Type == c.Type {
+			if condition.Status == c.Status &&
+				condition.Reason == c.Reason &&
+				condition.Message == c.Message {
+				return
+			}
+
+			found = true
+			conditions = append(conditions, *condition)
+		} else {
+			conditions = append(conditions, c)
+		}
+	}
+	if !found {
+		conditions = append(conditions, *condition)
+	}
+
+	ci.Status.Conditions = conditions
+}
+
+// SetStatusSyncCondition modifies the ClusterIngress object's status by setting
+// its SyncSuccessful condition in accordance with the provided error value.
+func (ci *ClusterIngress) SetStatusSyncCondition(err error) {
+	condition := &operatorapi.OperatorCondition{
+		Type:   operatorapi.OperatorStatusTypeSyncSuccessful,
+		Status: operatorapi.ConditionTrue,
+	}
+
+	if err != nil {
+		condition.Status = operatorapi.ConditionFalse
+		condition.Reason = "SyncFailed"
+		condition.Message = err.Error()
+	}
+
+	ci.setStatusCondition(condition)
+}
+
+// SetStatusAvailableCondition modifies the ClusterIngress object's status by
+// setting its Available condition to the specified status and message.
+func (ci *ClusterIngress) SetStatusAvailableCondition(available bool, message string) {
+	condition := &operatorapi.OperatorCondition{
+		Type:   operatorapi.OperatorStatusTypeAvailable,
+		Status: operatorapi.ConditionTrue,
+	}
+
+	if !available {
+		condition.Status = operatorapi.ConditionFalse
+		condition.Reason = "ServiceNotReady"
+		condition.Message = message
+	}
+
+	ci.setStatusCondition(condition)
+}
+
+// clusterIngressStatusEqual returns true if and only if the status conditions
+// (ignoring LastTransitionTime) and ingresses of the provided ClusterIngress
+// objects are equal.
+func clusterIngressStatusEqual(oldCi, ci *ClusterIngress) bool {
+	numConditions := len(ci.Status.Conditions)
+	if numConditions != len(oldCi.Status.Conditions) {
+		return false
+	}
+	for i := 0; i < numConditions; i++ {
+		conditionA := oldCi.Status.Conditions[i]
+		conditionB := ci.Status.Conditions[i]
+
+		// Compare every field except LastTransitionTime.
+		if conditionA.Type != conditionB.Type ||
+			conditionA.Status != conditionB.Status ||
+			conditionA.Reason != conditionB.Reason ||
+			conditionA.Message != conditionB.Reason {
+			return false
+		}
+	}
+
+	numIngresses := len(ci.Status.LoadBalancer.Ingress)
+	if numIngresses != len(oldCi.Status.LoadBalancer.Ingress) {
```

Comment:
Perform load balancer checks only when ci.Spec.HighAvailability.Type == ingressv1alpha1.CloudClusterIngressHA ?
When ci.Spec.HighAvailability.Type  != CloudClusterIngressHA , need to clear old Status.LoadBalancer field.

Reply:
As you point out in another comment, the logic in `syncServiceUpdate` needs to be fixed for `ci.Spec.HighAvailability.Type != CloudClusterIngressHA`, but once that is fixed, I think the equality check should still always compare `.Status.LoadBalancer.Ingress`.  If we ever support changing from `CloudClusterIngressHA` to some other value, we want the equality check to return false so that we update with the status with the empty list of ingresses.  Am I missing anything?